e2e/start/basic/app/routes/-server-fns/serialize-formdata-fn-call.tsx

```diff
@@ -4,29 +4,40 @@ import { createServerFn } from '@tanstack/start'
 const testValues = {
   name: 'Sean',
   age: 25,
+  pet1: 'dog',
+  pet2: 'cat',
   __adder: 1,
 }
 
 export const greetUser = createServerFn()
-  .validator((data: unknown) => {
+  .validator((data: FormData) => {
     if (!(data instanceof FormData)) {
       throw new Error('Invalid! FormData is required')
     }
     const name = data.get('name')
     const age = data.get('age')
+    const pets = data.getAll('pet')
 
-    if (!name || !age) {
-      throw new Error('Name and age are required')
+    if (!name || !age || pets.length === 0) {
+      throw new Error('Name, age and pets are required')
     }
 
     return {
       name: name.toString(),
       age: parseInt(age.toString(), 10),
+      pets: pets.map((pet) => pet.toString()),
     }
   })
-  .handler(async ({ data: { name, age } }) => {
-    return `Hello, ${name}! You are ${age + testValues.__adder} years old.`
-  })
+  .handler(
```

Comment:
Please switch this to using explicit returns with their brackets.
Also, don't restructure the pets array, instead just join with a comma.

```js
.handler(async ({ data: { name, age, pets } }) => {
	return `Hello, ${name}! You are ${age + testValues.__adder} years old, and your favourite pets are ${pets.join(',')}.`
})
```

Reply:
can do, but why, out of curiosity?

Reply:
the linting was complaining about `async` with no `await`, fwiw

Reply:
updated, sans async keyword

Reply:
> can do, but why, out of curiosity?

Legibility mostly. For a lib, the more legible the better.